### ISSUE TITLE
NSFW Content Filter addon does not filter own posts

### ISSUE THREAD

**Issue by SpencerDub** [friendica/friendica#8890]
### Expected behavior
When a filter is set in NSFW Content Filter, it will apply to all posts, including the user's own

### Actual behavior
NSFW Content Filter does not hide a user's own posts, even if they contain filtered terms

### Steps to reproduce the problem
1. Locate a post by another user in your stream to act as a reference post. Pick a term from that post.
2. Enable NSFW Content Filter. Add the term you selected to your filter list.
3. Compose a new post containing the chosen term.
4. Compare your new post to the reference post in your stream. The reference post is hidden; your own post is not.

### Friendica version you encountered the problem
Friendica 'Red Hot Poker' 2020.03 - 1338

### PHP version
7.0.33-0+deb9u8

### SQL version
10.1.44-MariaDB-0+deb9u1

### Additional comments
I suppose I could see an argument made for never hiding one's own posts--you wrote them, why would you not want to see them?--but I disagree with it. One can, for instance, create and share NSFW content and still not want any NSFW content, even their own, displayed prominently in their feed.

**Comment by MrPetovan** [comment]
This is an intended exception and I guess we can add an add-on user setting to control this behavior?

**Comment by MrPetovan** [comment]
I predict the issue will be that the exception is located in the core content filter engine, not in this specific addon (or any other providing content filtering implementations).

**Comment by iworkinpixels** [comment]
I agree with @SpencerDub on this... for instance, someone with an OnlyFans and a day job at an office might wish to view their feed at their day job whilst still having their adult content hidden by default.

How difficult would it be to add a setting to the core content filter engine to let the user choose whether to filter their own posts or not? I could take a crack at it and send in a pull request if that would be helpful?

**Comment by tobiasd** [comment]
> for instance, someone with an OnlyFans and a day job at an office might wish to view their feed at their day job whilst still having their adult content hidden by default.

For posting stuff? Things that you post publicly will be visible for everyone (e.g. community streams on the local and other servers). I'd put the _OnlyFans_ contacts in a special contact group and post stuff for them only into that group. Your other contacts wont see them, your public profile will not include them for stranger visitors and your boss wont get that postings as well (except if they are in that group).

**Comment by iworkinpixels** [comment]
Ok, but IMO that kind of misses the point... Yes, I could hide content by posting to a group, but that assumes that I want to limit every type of NSFW content I post to only groups, and that I am never in a situation where I don't want someone looking over my shoulder to see certain content.

On my FB feed I posted nearly everything to my main feed, from political activism to poly stuff to rants about the drivers in town, all sorts of memes, some nsfw and some not... and occasionally I had jobs which allowed me to use Facebook at work, but which still probably would have raised an eyebrow if they saw one of my shibari posts on my screen, or anything regarding my political party. I do not wish to limit my posts about all of these subjects to only groups, so it would be a useful feature for me to be able to post everything that I wish to post on main and still have the things which I have marked as nsfw to not automatically show up when on my feed. That way regardless of whether I post it to a group or not, I know I'm safe to open my feed at work and nobody will see anything offensive over my shoulder.

The same is true of when I'm on my cell phone and there are kids around, and therefore hiding such content might be desirable in those settings as well.

Also, it's useful as a diagnostic tool... if I were to post NSFW content to a group that contains both SFW and NSFW content, but I forgot to include "nsfw" in the post, or I was typing too fast and typed "nfsw" instead, then right now I get no visual feedback on my feed that something is wrong, I have to wait for one of my friends to notice... something which might get me in significant trouble in certain circumstances. It would be better if I hit save on the post, view my profile, and see that the post isn't auto-hiding and then I can correct the problem hopefully before anyone else sees the post.

**Comment by tobiasd** [comment]
> Ok, but IMO that kind of misses the point... Yes, I could hide content by posting to a group, but that assumes that I want to limit every type of NSFW content

I thought that was the reasoning behind the OnlyFans service. The platform never made it into my filter bubble, but the name suggested somehow that the postings you do there, are only visible to your fans.

**Comment by iworkinpixels** [comment]
Ahh, I see. The main content itself, yeah. But then in order to promote your site you have to advertise, usually by posting censored versions of the images to Instagram or Facebook or Twitter.

But that's off in the weeds. The point is people may wish to be able to open their friendica feed without people being able to view explicit content over their shoulder, even and perhaps especially if it's their own explicit content.
